parlai/tasks/wrapper/agents.py

```diff
@@ -144,3 +105,58 @@ def shutdown(self):
 
     def update_counters(self):
         self.task.update_counters()
+
+
+class LabelToTextTeacher(AbstractWrapperTeacher):
+    """
+    Teacher that will shift message['labels'][0] into message['text'] for whatever task
+    is specified with --label-to-text-task.
+
+    Because the dialogue history is effectively overwritten by this action, all episodes
+    will be flattened into one example each.
+    """
+
+    @classmethod
+    def add_cmdline_args(cls, argparser):
+        # Teacher has no args
+        parser = argparser.add_argument_group('LabelToText args')
+        parser.add_argument(
+            '-l2tt',
+            '--label-to-text-task',
+            type=str,
+            help='The task whose labels will get shifted into the text field',
+        )
+
+    def __init__(self, opt: Opt, shared=None):
+        if ',' in opt['task']:
+            raise ValueError('LabelToTextTeacher cannot be used with multiple tasks!')
+        self.id = opt['task']
+        self.opt = opt
+        if shared and 'task' in shared:
```

Comment:
when would 'task' not be in shared? should this raise an error?

Reply:
Good point, we could remove `'task' in shared` and possibly raise an assert (or just let it fail if 'task' is not in shared)